graylog2-web-interface/src/components/alertnotifications/AlertNotification.jsx

```diff
@@ -35,13 +37,21 @@ const AlertNotification = React.createClass({
 
   _onSubmit(data) {
     AlarmCallbacksActions.update(this.props.alertNotification.stream_id, this.props.alertNotification.id, data)
-      .then(() => AlertNotificationsActions.listAll());
+      .then(() => {
+        if (typeof this.props.onNotificationUpdate === 'function') {
```

Comment:
We are very inconsistent with that, but I think it is safer to ensure it is a function when using it. The prop type check may not always be there (for instance if we ever enable the production mode in react), and it only logs a warning, so it will not prevent the function to be called, and the application would be broken in that case.